hub/core/transform/transform_shard.py

```diff
@@ -0,0 +1,44 @@
+from hub.core.transform.transform_tensor import TransformDatasetTensor
+from hub.util.exceptions import TensorDoesNotExistError
+
+# TODO: add tests
+
+
+def slice_with_list(item, slice_list):
+    """Slice an object with a list of slices."""
+    sliced = item
+    for slice_ in slice_list:
+        sliced = sliced[slice_]
+    return sliced
+
+
+class TransformDatasetShard:
```

Comment:
I'm using this class with interface very similar to Dataset to avoid chunk engine overhead while storing samples during transforms. The append, extend and numpy syntax should be very similar to Dataset currently, but might not work for advanced indexing.

Reply:
Need to add docstrings

Reply:
can we try to avoid `Shard` naming? it's kind of confusing

Reply:
any suggestions?

Reply:
none come to mind, but i feel like we use `Shard` and `Chunk` interchangably

Reply:
shard here is more applicable though than chunk